tests/perftest.py

```diff
@@ -16,6 +16,51 @@
 import utils
 from phonetest import PhoneTest, PhoneTestResult
 
+"""
+   PerfherderArtifact and PerfherderSuite and specific formats for
+   Perfherder as defined in:
+   https://bugzilla.mozilla.org/show_bug.cgi?id=1175295
+
+   These work with summarized suite values and summarized test value,
+   instead of working with raw replicates.
+
+   In the future it would be nice to have a generic PerfData class which
+   stores the raw replicates and then the output functions can do the final
+   summarization and calculations as defined by the output medium (e.g. perfherder)
+"""
+class PerfherderArtifact():
+    def __init__(self, suites=[]):
+        self._framework = 'autophone'
+        self._suites = suites
+
+    def addSuite(self, suite):
+        self._suites.append(suite)
+
+    def toJSON(self):
+        return {'framework': self._framework,
+               'suites': [s.toJSON() for s in self._suites]}
+
+
+class PerfherderSuite():
+    def __init__(self, name=None, value=0, subtests=[]):
+        self._name = name
+        self._value = value
+        self._subtests = subtests
+
+    def addSubTest(self, subtest):
+        if 'name' not in subtest or 'value' not in subtest:
+            raise ValueError("%s is not a value performance suite subtest" % subtest)
+        self._subtests.append(subtests)
+
+    def addSubTest(self, name, value):
+        self._subtests.append({'name': name, 'value': value})
+
+    def toJSON(self):
```

Comment:
Technically this isn't json, it's a python dictionary. :)

It would probably be easiest to just derive this class from a dictionary, in fact:

    class PerfherderSuite(dict):
        
        def __init__(self, name=None, value=0, subtests=None):
            if subtests is None:
                self['subtests'] = []
            else:
                self['subtests'] = subtests
            self['name'] = name
            self['value'] = value
            
        def add_subtest(self, name, value):
            self['subtests'].append({'name': name, 'value': value})
            
    ps = PerfherderSuite(name='tps', value=123, subtests=[{'name': 'tps1', 'value': 1234}])
    print ps
    ps.add_subtest('tps2', 12345)
    print ps